js/source/source_cache.js

```diff
@@ -255,7 +262,15 @@ class SourceCache extends Evented {
             }
             if (this._cache.has(coord.id)) {
                 retain[coord.id] = true;
-                return this._cache.get(coord.id);
+                const cachedTile = this._cache.get(coord.id);
+
+                if (this._cacheTimers[coord.id]) {
+                    clearTimeout(this._cacheTimers[coord.id]);
+                    delete this._cacheTimers[coord.id];
```

Comment:
Oo @anandthakker thank you — did not know this, googled a little hole for myself to understand further: https://github.com/google/google-api-nodejs-client/issues/375